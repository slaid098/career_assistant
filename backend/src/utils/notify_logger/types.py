from abc import ABC, abstractmethod
from typing import Any, Literal, Protocol, TypedDict

LogLevel = Literal["DEBUG", "INFO", "SUCCESS", "WARNING", "ERROR", "CRITICAL"]

CompressionType = Literal[
    "zip",
    "tar",
    "gz",
    "bz2",
    "xz",
    "lzma",
    "tar.gz",
    "tar.bz2",
    "tar.xz",
]


class LogHandler(ABC):
    @property
    def name(self) -> str:
        return self.__class__.__name__

    @abstractmethod
    def add(self) -> None:
        raise NotImplementedError


class LogMessage(TypedDict):
    level: str
    timestamp: str
    message: str


class WebSocketClient(Protocol):
    """Protocol for WebSocket clients used by WebSocketHandlerManager."""

    async def send_json(self, data: Any) -> None:
        """Sends JSON-serializable data to the client"""
        ...
