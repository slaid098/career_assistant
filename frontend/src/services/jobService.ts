import { Job } from './jobService.types';

const getApiUrl = (): string => {
    const url = process.env.NEXT_PUBLIC_API_URL;
    if (!url) {
        throw new Error('NEXT_PUBLIC_API_URL is not defined');
    }
    return url;
};

const mockJobs: Job[] = [
    {
        id: 1,
        title: 'Frontend Developer (React)',
        company: 'Meta',
        url: '#',
        created_at: new Date().toISOString(),
        description: 'We are looking for a talented Frontend Developer to join our team. You will be responsible for building the next generation of our user interfaces.',
    },
    {
        id: 2,
        title: 'Fullstack Engineer',
        company: 'Google',
        url: '#',
        created_at: new Date(Date.now() - 86400000 * 2).toISOString(), // 2 days ago
        description: 'Join our team to work on exciting new projects that reach millions of users. Strong experience with Node.js and React is required.',
    },
    {
        id: 3,
        title: 'TypeScript Developer',
        company: 'Microsoft',
        url: '#',
        created_at: new Date(Date.now() - 86400000 * 5).toISOString(), // 5 days ago
        description: 'Are you passionate about TypeScript? We are looking for an expert to help us build robust and scalable applications.',
    },
];

const validateResponse = async <T>(response: Response): Promise<T> => {
    if (!response.ok) {
        const errorText = await response.text();
        console.error('API Error:', errorText);
        throw new Error(`Network response was not ok: ${response.statusText}`);
    }
    return response.json() as Promise<T>;
};

export const getJobs = async (): Promise<Job[]> => {
    const getJobUrl = `${getApiUrl()}/jobs`;
    try {
        const response = await fetch(getJobUrl);
        return validateResponse<Job[]>(response);
    } catch (error) {
        console.warn('Could not fetch jobs from backend, returning mock data.', error);
        return mockJobs;
    }
};

export const getJobById = async (id: string): Promise<Job | null> => {
    const getJobUrl = `${getApiUrl()}/jobs/${id}`;
    try {
        const response = await fetch(getJobUrl);
        if (response.status === 404) {
            return null; // Explicitly handle 404 Not Found
        }
        return validateResponse<Job>(response);
    } catch (error) {
        console.warn(`Could not fetch job ${id} from backend, returning mock data.`, error);
        const job = mockJobs.find((j) => j.id.toString() === id) || null;
        return job;
    }
};





