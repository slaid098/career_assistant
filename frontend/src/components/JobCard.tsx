import React from 'react';
import { Job } from '@/services/jobService.types';

type JobCardProps = {
  job: Job;
};

const JobCard: React.FC<JobCardProps> = ({ job }) => {
  return (
    <div>
      <h2>{job.title}</h2>
      <p>{job.company}</p>
    </div>
  );
};

export default JobCard; 