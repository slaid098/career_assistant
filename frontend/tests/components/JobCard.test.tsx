import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import JobCard from '@/components/JobCard';
import { Job } from '@/services/jobService.types';

describe('JobCard component', () => {
  const mockJob: Job = {
    id: 1,
    title: 'Senior Software Engineer',
    company: 'Innovate Inc.',
    url: 'https://example.com/job/1',
    created_at: new Date().toISOString(),
  };

  it('should render the job title and company name', () => {
    render(<JobCard job={mockJob} />);

    expect(screen.getByText('Senior Software Engineer')).toBeInTheDocument();
    expect(screen.getByText('Innovate Inc.')).toBeInTheDocument();
  });
}); 