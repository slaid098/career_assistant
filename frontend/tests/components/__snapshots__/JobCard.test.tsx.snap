// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`JobCard component should match the snapshot 1`] = `
<div>
  <a
    class="block bg-white p-6 border border-gray-200 rounded-lg shadow-sm hover:shadow-lg transition-all duration-300 ease-in-out transform hover:-translate-y-1"
    href="/jobs/1"
  >
    <div
      class="flex items-start"
    >
      <div
        class="flex-shrink-0 mr-4"
      >
        <img
          alt="Innovate Inc. logo"
          class="rounded-full"
          data-nimg="1"
          decoding="async"
          height="56"
          loading="lazy"
          src="/globe.svg"
          style="color: transparent;"
          width="56"
        />
      </div>
      <div
        class="flex-grow"
      >
        <h2
          class="text-xl font-bold text-gray-800"
        >
          Senior Software Engineer
        </h2>
        <p
          class="text-md text-gray-600 mt-1"
        >
          Innovate Inc.
        </p>
        <p
          class="text-sm text-gray-500 mt-2"
        >
          Posted on 
          July 6, 2025
        </p>
      </div>
    </div>
  </a>
</div>
`;
